index.html

```diff
@@ -159,141 +156,401 @@ <h2>
-              it is sufficient for a subsequent {{requestPointerLock()}}.
-            </p>
-          </aside>
-          <p>
-            If any element (including this one), whose <a>shadow-including
-            root</a> is same as this element's <a>shadow-including root</a>, is
-            already locked (or pending lock) the pointer lock <a>target</a>
-            must be updated to this element and a <a>pointerlockchange</a>
-            event sent.
-          </p>
-          <p>
-            If any element, whose <a>shadow-including root</a> is a different
-            document, is already locked the request must fail and a
-            <a>pointerlockerror</a> event be sent.
-          </p>
-          <p>
-            Once in the locked state the <a>user agent</a> must fire all
-            relevant user generated {{MouseEvent}} events (for example:
-            `mousemove`, `mousedown`, `mouseup`, `click`, and `wheel`)
-            [[ui-events]] to the <a>target</a> of pointer lock, and not fire
-            mouse events to other elements. Events that require the concept of
-            a mouse cursor must not be dispatched (for example: `mouseover`,
-            `mouseout`, `drag`, and `drop`).
-          </p>
-          <p>
-            In the locked state the system mouse cursor must be hidden.
-            Movement and button presses of the mouse must not cause the window
-            to lose focus.
-          </p>
-          <p>
-            Synthetic mouse events created by application script act the same
-            regardless of lock state.
+            <br>
+            When <dfn>requestPointerLock</dfn> invoked, perform the following
+            steps:
           </p>
+          <ol>
+            <li>Let <var>promise</var> be [=a new promise=].
+            </li>
+            <li>When a window is in focus, if the [=this=]'s [=shadow-including
+            root=]is the [=navigable/active document=] of a [=Document/browsing
+            context=] (or has an ancestor browsing context) which is not in
+            focus:
+              <ol>
+                <li>[=Queue an element task=] on the [=user interaction task
+                source=], given this, to perform the following steps:
+                  <ol>
+                    <li>[=Fire an event=] named {{pointerlockerror}} at
+                    [=this=]'s [=Node/node document=].
+                    </li>
+                    <li>[=Reject=] <var>promise</var> with a
+                    "{{WrongDocumentError}}" {{DOMException}}.
+                    </li>
+                  </ol>
+                </li>
+                <li>Return <var>promise</var>.
+                </li>
+              </ol>
+            </li>
+            <li>If the user agent's [=pointer-lock target=] is an element whose
+            [=shadow-including root=]is not equal to [=this=]'s
+            [=shadow-including root=], then:
```

Comment:
We are also a bit unsure what role the [=shadow-including root=] plays here? Can you explain the rationale for this?

Reply:
also, anything actually testable here?

Reply:
File https://github.com/w3c/pointerlock/issues/92